File: app/src/main/java/at/shockbytes/corey/ui/fragment/workoutpager/ExercisePagerFragment.java
package at.shockbytes.corey.ui.fragment.workoutpager;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import at.shockbytes.corey.R;
import at.shockbytes.corey.common.core.workout.model.Exercise;
import butterknife.BindView;
import butterknife.ButterKnife;


public class ExercisePagerFragment extends Fragment {

    private static final String ARG_EXERCISE = "arg_exercise";

    private Exercise exercise;

    @BindView(R.id.fragment_pageritem_exercise_txt)
    protected TextView text;

    public ExercisePagerFragment() {
    }

    public static ExercisePagerFragment newInstance(Exercise exercise) {
        ExercisePagerFragment fragment = new ExercisePagerFragment();
        Bundle args = new Bundle();
        args.putParcelable(ARG_EXERCISE, exercise);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        exercise = getArguments().getParcelable(ARG_EXERCISE);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_pageritem_exercise, container, false);
        ButterKnife.bind(this, v);
        return v;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        text.setText(exercise.getDisplayName(getContext()));
    }
}

File: app/src/main/java/at/shockbytes/corey/ui/fragment/workoutpager/ExercisePagerFragment.kt
package at.shockbytes.corey.ui.fragment.workoutpager

import android.os.Bundle
import android.widget.TextView
import at.shockbytes.core.ui.fragment.BaseFragment
import at.shockbytes.corey.R
import at.shockbytes.corey.common.core.workout.model.Exercise
import at.shockbytes.corey.dagger.AppComponent
import kotterknife.bindView

class ExercisePagerFragment : BaseFragment<AppComponent>() {

    override val snackBarBackgroundColorRes: Int = R.color.sb_background
    override val snackBarForegroundColorRes: Int = R.color.sb_foreground

    private var exercise: Exercise? = null

    private val text: TextView by bindView(R.id.fragment_pageritem_exercise_txt)

    override val layoutId = R.layout.fragment_pageritem_exercise

    override fun injectToGraph(appComponent: AppComponent?) = Unit

    override fun setupViews() {
        text.text = exercise?.getDisplayName(context!!)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        exercise = arguments?.getParcelable(ARG_EXERCISE)
    }

    override fun bindViewModel() = Unit

    override fun unbindViewModel() = Unit

    companion object {

        private const val ARG_EXERCISE = "arg_exercise"

        fun newInstance(exercise: Exercise): ExercisePagerFragment {
            val fragment = ExercisePagerFragment()
            val args = Bundle()
            args.putParcelable(ARG_EXERCISE, exercise)
            fragment.arguments = args
            return fragment
        }
    }
}
